docs/src/main/asciidoc/http-reference.adoc

```diff
@@ -16,28 +16,27 @@ include::./attributes.adoc[]
 :sectnumlevels: 4
 
 
-This document explains various Undertow features that you can use in Quarkus.
+This document explains various HTTP features that you can use in Quarkus.
+
+HTTP is provided using Eclipse Vert.x as the base HTTP layer. Servlet's are supported using a modified version of Undertow that
+runs on top of Vert.x, and RESTeasy is used to provide JAX-RS support. If Undertow is present RESTeasy will run as a
+Servlet filter, otherwise it will run directly on top of Vert.x with no Servlet involvement.
 
 == Serving Static Resources
 
-To serve static resources you must place them in the `META-INF/resources` directory of your application. This is because
-Quarkus applications are jar files rather than war files, and this location is the standard location for resources in
-jar files as defined by the Servlet spec.
+To serve static resources you must place them in the `META-INF/resources` directory of your application. This location
+was chosen as it is the standard location for resources in `jar` files as defined by the Servlet spec. Even though
+Quarkus can be used without Servlet following this convention allows existing code that places its resources in this
+location to function correctly.
 
 == Configuring the Context path
 
-By default Undertow will serve content from under the root context. If you want to change this you can use the
-`quarkus.servlet.context-path` config key to set the context path.
-
-== undertow-handlers.conf
-
-You can make use of the Undertow predicate language using an `undertow-handlers.conf` file. This file should be placed
-in the `META-INF` directory of your application jar. This file contains handlers defined using the
-link:http://undertow.io/undertow-docs/undertow-docs-2.0.0/index.html#predicates-attributes-and-handlers[Undertow predicate language].
+By default Quarkus will serve content from under the root context. If you want to change this you can use the
+`quarkus.http.root-path` config key to set the context path.
 
-== web.xml
-
-If you are using a `web.xml` file as your configuration file, you can place it in the `src/main/resources/META-INF` directory.
+If you are using Servlet you can control the Servlet context path via `quarkus.servlet.context-path`. This item is relative
+to the http root above, and will only affect Servlet and things that run on top of Servlet. Most applications will
+want to use the HTTP root as this affects everything that Quarkus serves.
```

Comment:
I am confused here about how the two combine together. What happens when someone specifies both?

Reply:
It's actually explained in the javadoc of the config itself, but should be clarified here.

Reply:
it's explained in the next paragraph.

Reply:
I added that after @FroMage complained :-)